pageserver/src/bin/pageserver.rs

```diff
@@ -486,23 +509,49 @@ fn start_pageserver(conf: &'static PageServerConf) -> Result<()> {
             endpoint::serve_thread_main(router, http_listener)
         })?;
 
-    join_handles.push(http_endpoint_thread);
-
     // Spawn a thread to listen for connections. It will spawn further threads
     // for each connection.
     let page_service_thread = thread::Builder::new()
-        .name("Page Service thread".into())
+        .name("1 Page Service thread".into())
         .spawn(move || {
             page_service::thread_main(conf, auth, pageserver_listener, conf.auth_type)
         })?;
 
-    join_handles.push(page_service_thread);
-
-    for handle in join_handles.into_iter() {
-        handle
-            .join()
-            .expect("thread panicked")
-            .expect("thread exited with an error")
+    for info in &mut signals {
+        info!("Received a signal {:?}", info);
+        match info.signal {
+            SIGQUIT => {
+                info!("Got SIGQUIT. Terminate pageserver in immediate shutdown mode");
+                // TODO Do we need any actions here?
+                exit(1);
```

Comment:
I wonder, if it makes sense to return `111` or whatever else less generic to indicate a subtle difference between "just a generic failure" and a forced exit case?